features/networking/service.feature

```diff
@@ -154,3 +154,233 @@ Feature: Service related networking scenarios
+    Given I register clean-up steps:
+    """
+    Given as admin I successfully merge patch resource "networks.config.openshift.io/cluster" with:
+      | {"spec":{"externalIP":{"policy": {"allowedCIDRs": null, "rejectedCIDRs": null}}}}  |
+    """
+
+    # Create a svc with externalIP/22.2.2.10 which is in 22.2.2.0/25
+    Given I have a project 
+    And I wait up to 300 seconds for the steps to pass:
+    """
+    When I run oc create over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/externalip_service1.json" replacing paths:
+      | ["spec"]["externalIPs"][0]  | 22.2.2.10 |
+    Then the output should contain "externalIP is not allowed"
+    """
+
+    # Create a svc with externalIP/22.2.2.130 which is not in 22.2.2.0/25
+    And I wait up to 300 seconds for the steps to pass:
+    """
+    When I run oc create over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/externalip_service1.json" replacing paths:
+      | ["spec"]["externalIPs"][0]  | 22.2.2.130 |
+    Then the step should succeed
+    """
+ 
+    # Create a pod
+    When I run the :create client command with:
+      | f | https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/routing/caddy-docker.json |
+    Then the step should succeed
+    And the pod named "caddy-docker" becomes ready
+ 
+    # Curl externalIP:portnumber on new pod 
+    When I execute on the pod:
+      | /usr/bin/curl | -k | 22.2.2.130:27017 |
+    Then the output should contain:
+      | Hello-OpenShift-1 http-8080           |
+
+  # @author weliang@redhat.com
+  # @case_id OCP-24670
+  @admin
+  @destructive
+  Scenario: externalIP defined in service with set ExternalIP in rejectedCIDRs
+    Given I have a project
+    And SCC "privileged" is added to the "system:serviceaccounts:<%= project.name %>" group
+    Given I store the schedulable nodes in the :nodes clipboard
+    And the Internal IP of node "<%= cb.nodes[0].name %>" is stored in the :hostip clipboard
+   
+    # Create additional network through CNO
+    Given as admin I successfully merge patch resource "networks.config.openshift.io/cluster" with:
+      | {"spec":{"externalIP":{"policy":{"rejectedCIDRs":["<%= cb.hostip %>/24"]}}}} |
+ 
+    # Clean-up required to erase above externalIP policy after testing done
+    Given I register clean-up steps:
+    """
+    Given as admin I successfully merge patch resource "networks.config.openshift.io/cluster" with:
+      | {"spec":{"externalIP":{"policy":{"rejectedCIDRs": null}}}} |
+    """
+
+    # Create a svc with externalIP
+    Given I switch to the first user
+    And I wait up to 300 seconds for the steps to pass:
+    """
+    When I run oc create over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/externalip_service1.json" replacing paths:
+      | ["spec"]["externalIPs"][0]  | <%= cb.hostip %> |
+    And the output should contain "externalIPs have been disabled"
```

Comment:
Same here.

Reply:
@liangxia Add "Then the step should fail" check step and remove message checking steps. Thanks!